src/averageAges.js

```diff
@@ -37,7 +41,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = !withChildren
+    ? people.filter((user) => user.sex === 'f')
+    : people.filter((user) =>
+      user.sex === 'f' && people.some((child) => user.name === child.mother));
```

Comment:
will be easier to read if you create a variable to check existing children on women